### PR TITLE
ci/cd: open-source compliance - copyright headers check

### PR DESCRIPTION
**Pull request by GhiathAjam**
Add an action checker for copyright headers for all current applicable file types.

- [x] .yml files
- [x] .cfg files
- [x] .md files
- [x] .py files
- [x] .toml files
- [x] requirements files
- [x] .typed == python typing files
- [x] .env.example files
- [x] .gitignore

### CODE REVIEW

**Comment by GhiathAjam** [comment]
Merging #27 will fix the pipeline of this branch

**Review by albert17 — APPROVED** [review]

File: .github/workflows/copy_right_headers_check.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Copyright (c) Microsoft Corporation.
```
[the remark this answers is not in the document]
Comment (GhiathAjam) [reply .github/workflows/copy_right_headers_check.yml:1]: I did check for Python but couldn't find anything. I could dig up some more but figured the repo contains a lot more types than `.py` files and checking with `grep` works on pretty much anything.

**Comment by EssamWisam** [comment]
Just came in here to say this is top-of-the-line work @GhiathAjam .

**Comment by GhiathAjam** [comment]
My salutes to you, Mr. @EssamWisam.

**Pull request merged**